plugins/builds/steps/logs.js

```diff
@@ -207,20 +207,36 @@ module.exports = config => ({
                             })
                         )
                         .then(([lines, morePages]) => {
-                            if (type !== 'download') {
-                                return h.response(lines).header('X-More-Data', (morePages || !isDone).toString());
-                            }
+                            if (type === 'download') {
+                                let res = '';
 
-                            let res = '';
+                                for (let i = 0; i < lines.length; i += 1) {
+                                    res = `${res}${lines[i].m}\n`;
+                                }
 
-                            for (let i = 0; i < lines.length; i += 1) {
-                                res = `${res}${lines[i].m}\n`;
+                                return h
+                                    .response(res)
+                                    .type('text/plain')
+                                    .header('content-disposition', `attachment; filename="${stepName}-log.txt"`);
                             }
 
-                            return h
-                                .response(res)
-                                .type('text/plain')
-                                .header('content-disposition', `attachment; filename="${stepName}-log.txt"`);
+                            const formatedLine = lines.map(line => {
```

Comment:
I think looping through is  inefficient, using a regex over the string is what comes to my mind, and regex optimization is an already understood problem.